### PR TITLE
Split test_excel.py (#24472)

### LINKED ISSUE #24472: TST: Split test_excel into sub test files
Logical follow up to https://github.com/pandas-dev/pandas/pull/24423/files#r244284618

cc @gfyoung

### PR DESCRIPTION
**Pull request by stevenbw**
Split tests/io/test_excel.py into two files, base.py and xlrd.py located in tests/io/excel/. The previous file was very large and a logical split has been applied to separate the `xlrd` tests.

- [ ] closes #24472 
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by simonjayhawkins** [comment]
@stevenbw The files will need to be named test_base.py and test_xlrd.py

File: pandas/tests/io/excel/test_base.py [hunk 1]
```diff
@@ -128,96 +77,6 @@ def set_engine(self, request):
         yield
         setattr(self, func_name, old_func)
 
-    @td.skip_if_no("xlrd", "1.0.1")  # see gh-22682
-    def test_usecols_int(self, ext):
```
Comment (simonjayhawkins) [inline pandas/tests/io/excel/test_base.py]: you'll probably just want to split by complete classes in this PR and leave refactoring the classes to a follow-on PR.

File: pandas/tests/io/excel/test_xlrd.py [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+_tsd = tm.getTimeSeriesData()
+_frame = DataFrame(_seriesd)[:10]
+_frame2 = DataFrame(_seriesd, columns=['D', 'C', 'B', 'A'])[:10]
+_tsframe = tm.makeTimeDataFrame()[:5]
+_mixed_frame = _frame.copy()
+_mixed_frame['foo'] = 'bar'
+
+
+@contextlib.contextmanager
+def ignore_xlrd_time_clock_warning():
```
Comment (simonjayhawkins) [inline pandas/tests/io/excel/test_xlrd.py]: any code required by both `test_base.py` and `test_xlrd.py` should probably be in `test_base.py`

File: pandas/tests/io/excel/test_xlrd.py [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+        df : DataFrame
+        """
+        pth = os.path.join(self.dirpath, basename + ext)
+        return read_excel(pth, *args, **kwds)
+
+
+class XlrdReadingTestsBase(XlrdSharedItems):
+    # This is based on ExcelWriterBase
+
+    @pytest.fixture(autouse=True, params=['xlrd'])
```
Comment (simonjayhawkins) [inline pandas/tests/io/excel/test_xlrd.py]: code duplicated from test_base

File: pandas/tests/io/excel/test_base.py [hunk 4]
```diff
@@ -80,46 +64,11 @@ def get_csv_refdf(self, basename):
...
-        """
-        pth = os.path.join(self.dirpath, basename + ext)
-        return read_excel(pth, *args, **kwds)
-
 
 class ReadingTestsBase(SharedItems):
     # This is based on ExcelWriterBase
 
-    @pytest.fixture(autouse=True, params=['xlrd', None])
+    @pytest.fixture(autouse=True, params=[None])
```
Comment (simonjayhawkins) [inline pandas/tests/io/excel/test_base.py]: maybe a better name for `test_base` would be `test_common` or `test_all_engines`. this split is creating too much code duplication. maybe a split on reader/writer tests first before splitting out xlrd

**Comment by WillAyd** [comment]
Agreed high level with @simonjayhawkins think there are just too many things moved to `test_xlrd`. If anything the entire `SharedItems` class you have can remain in the base file.

Did this yield the same number of tests before and after?

**Review by jreback — CHANGES_REQUESTED** [review]
so before this is actually split; it needs to be fixturized a bit more I think. not really sure the best way of doing this, but if you are just moving files, then that can be in a single PR. We don't want to mix moving things with changing things as its very hard to tell what is going on. So first move w/o any changes, then new PR' to change.

**Comment by stevenbw** [comment]
Thanks for the support @simonjayhawkins, @WillAyd, @jreback :). I have taken on board your comments and attempted to simply split the file without separating classes or duplicating code. However, due to classes relying on previous classes, for example `TestXlrdReader`, which could belong to `test_xlrd.py` requires `ReadingTestsBase`,  which belongs to `test_all_engines.py`, there is only a small amount of code which can be moved. Other than `TestXlrdReader` there is only `ignore_xlrd_time_clock_warning` which could be in a separate file but it is used several times in `ReadingTestsBase`, it could be imported but doesn't seem to be of much benefit.

Unless I am missing something, it seems the ideal solution would be to rewrite the classes so they can be decoupled. I appreciate this would be a big change though.

**Comment by simonjayhawkins** [comment]
@stevenbw the biggest classes by far are `ReadingTestsBase` and `TestExcelWriter`. i think these would need to be in separate files in order to make an impact on reducing the test file size. could the classes be grouped for this to work?

**Comment by stevenbw** [comment]
@simonjayhawkins Yes they could. They both have a fundamental dependency on `SharedItems` but one option could be to import this class, then have one file `test_reader.py` for `ReadingTestsBase` and `TestXlrdReder`. The other file could contain all other classes and be called `test_writer.py`.

**Comment by WillAyd** [comment]
Can you merge master here? Also we just refactored the excel code in #25153 so you may be able to mimic that set up on the test side

**Comment by stevenbw** [comment]
Yes I will take a look, should be able to mimic what you have done.

**Comment by WillAyd** [comment]
@stevenbw I think this is a duplicate of #25334 but if that's not the case let me know.

Generally should just keep pushing updates to the same PR rather than opening a new one

**Pull request closed without merging**